music21/musicxml/m21ToXml.py

```diff
@@ -1735,6 +1736,128 @@ def parseFlatScore(self):
+        the voiceNumberOffset needed in each part.  Parts that aren't in a
+        joinable group get offset 0.
+
+        Called automatically by .parse()
+        '''
+        partExporterForPart: t.Dict[stream.Part, PartExporter] = {}
+
+        # Compute partExp.staffGroup and partExp.maxNumVoices
+        for partExp in self.partExporterList:
+            partExporterForPart[partExp.stream] = partExp
+
+            joinableGroup = None
+            for sg in self.groupsToJoin:
+                if partExp.stream in sg:
+                    joinableGroup = sg
+                    break
+
+            partExp.staffGroup = joinableGroup
+            if partExp.staffGroup is None:
+                continue
+
+            # part is in a staffGroup, so we need to find the maximum voice
+            # number used in each measure in this part.  The maxVoiceNum
+            # for a measure is the higher of the number of voices and the
+            # maximum "low enough to not be considered a memory location"
+            # voice id.
+            for measure in partExp.stream[stream.Measure]:
+                voices = measure[stream.Voice]
+                maxNonMemLocationVoiceId: int = 0
+                for voice in voices:
+                    if (isinstance(voice.id, int)
+                            and voice.id < defaults.minIdNumberToConsiderMemoryLocation):
+                        maxNonMemLocationVoiceId = max(maxNonMemLocationVoiceId, voice.id)
+
+                # maxVoiceNum is 1-based (voice numbers are 1-based in MusicXML)
+                partExp.maxVoiceNumForMeasure[measure] = max(len(voices), maxNonMemLocationVoiceId)
+
+        # Compute partExp.voiceNumberOffsetForMeasure from the maxVoiceNumForMeasure
+        # of each measure "above" us in the staff group.
+        for partExp in self.partExporterList:
+            if partExp.staffGroup is None:
+                continue
+
+            staffGroupStreamList = partExp.staffGroup.getSpannedElements()
+            partIndexInGroup = staffGroupStreamList.index(partExp.stream)
+
+            for simultaneousMeasures in self.zip_measures(staffGroupStreamList):
+                # add up the voice number offsets contributed by each measure before
+                # (above) us in the staff group
+                measure = simultaneousMeasures[partIndexInGroup]
+                if measure is None:
+                    continue
+                voiceNumberOffset = 0
+                for i in range(0, partIndexInGroup):
+                    m = simultaneousMeasures[i]
+                    if m is None:
+                        continue
+                    pex = partExporterForPart[staffGroupStreamList[i]]
+                    voiceNumberOffset += pex.maxVoiceNumForMeasure[m]
+                partExp.voiceNumberOffsetForMeasure[measure] = voiceNumberOffset
+
+    @staticmethod
+    def zip_measures(streams: t.List[stream.Stream]):
```

Comment:
```suggestion
    def zip_measures(streams: t.Iterable[stream.PartStaff]):
```

Reply:
This doesn't work, because t.Iterable is not Sized, so you can't call len() on it.  (That's weird, why not?)

Reply:
Ah, so `Sequence`?

Reply:
🤦🏻‍♂️ Yes.  Pushed again.